addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,463 @@
+        for tax_detail in tax_details.values():
+            line_code = tax_detail["tax"].tax_group_id.id
+            l10n_pk_tax_detail = l10n_pk_tax_details.get(line_code, {})
+            l10n_pk_tax_detail.update(dict.fromkeys(("rate", "amount", "amount_currency"), 0.0))
+            l10n_pk_tax_detail["rate"] += tax_detail["tax"].amount
+            l10n_pk_tax_detail["amount"] += tax_detail["tax_amount"]
+            l10n_pk_tax_detail["amount_currency"] += tax_detail["tax_amount_currency"]
+            l10n_pk_tax_details[line_code] = l10n_pk_tax_detail
+        return l10n_pk_tax_details
+
+    # ================================ API methods ===========================
+
+    @api.model
+    def _l10n_pk_edi_no_config_response(self):
+        return {
+            'statusCode': 0,
+            'errorMessage': _(
+                "A token is still needs to be set or it's wrong for the E-invoice(Pakistan). "
+                "It needs to be added and verify in the Settings."
+            )
+        }
+
+    @api.model
+    def _l10n_pk_edi_get_token(self, company):
+        return company.sudo().l10n_pk_edi_token or False
+
+    @api.model
+    def _l10n_pk_edi_connect_to_server(self, company, payload, use_timeout=False):
+        timeout = self.env["ir.config_parameter"].sudo().get_param(
+            "l10n_pk_edi.time_out",  # While using proxy server it may take longer than 25 sec
+            use_timeout or DEFAULT_TIMEOUT
+        )
+        proxy_address = company.sudo().l10n_pk_edi_proxy
+        if not proxy_address:
+            raise UserError(_("Please add a proxy server from configuration to test APIs"))
+        proxies = {
+            'http': 'http://' + proxy_address,
+            'https': 'http://' + proxy_address,
+        }
+        headers = {
+            'Authorization': f"Bearer {company.sudo().l10n_pk_edi_token}",
+            'Content-Type': 'application/json',
+        }
+        try:
+            response = requests.post(
+                DEFAULT_ENDPOINT,
+                json=payload,
+                headers=headers,
+                proxies=proxies,
+                timeout=timeout
+            )
+            response.raise_for_status()
+            return response.json()
+        except (requests.exceptions.ConnectTimeout, requests.exceptions.ConnectionError) as e:
+            _logger.warning("Connection error: %s", e.args[0])
+            return {
+                'statusCode': 404,
+                'errorMessage': _("Unable to connect to the online E-invoice(Pakistan) service. "
+                    "The web service may be temporary down. Please try again in a moment."
+                )
+            }
+
+    @api.model
+    def _l10n_pk_edi_generate(self, company, json_payload):
+        token = self._l10n_pk_edi_get_token(company)
+        if not token:
+            return self._l10n_pk_edi_no_config_response()
+        return self._l10n_pk_edi_connect_to_server(company=company, payload=json_payload)
```

Comment:
```suggestion
        return self._l10n_pk_edi_connect_to_server(company, json_payload)
```
In this case, variable names are pretty self explanatory, and the order is already correct